packages/ui5-info/src/commandRunner.ts

```diff
@@ -0,0 +1,37 @@
+import { spawn } from 'child_process';
+
+/**
+ *
+ */
+export class CommandRunner {
```

Comment:
Instead of spawning a process we could also do an HTTP call to
https://registry.npmjs.org/@sapui5/distribution-metadata/ to list the versions, see https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#package-endpoints.

In this case we would need to allow to pass the host to the registry (default to 'https://registry.npmjs.org') in case the consumer uses an own NPM registry.

If we stick to `spawn`, I would not create `CommandRunner` as a generic class that allows to execute any command, but provide a function to run the exact call we need, like

```typescript
export function getUi5NpmVersions(): Promise<string[]> {
    return new Promise((resolve, reject) => {
        const stack: any = [];
        const cmd = process.platform === 'win32' ? 'npm.cmd' : 'npm';
        const args = ['show', '@sapui5/distribution-metadata', 'versions', '--no-color'];
        const spawnedCmd = spawn(cmd, args, {});
        spawnedCmd.stdout.setEncoding('utf8');
        let response: string;
        spawnedCmd.stdout.on('data', (data: Buffer) => {
            response = data.toString();
        });
        spawnedCmd.stderr.on('data', (data) => {
            stack.push(data.toString());
        });
        spawnedCmd.on('error', (error) => {
            reject(new Error(`Command failed with error: ${error.message}`));
        });
        spawnedCmd.on('close', (errorCode) => {
            if (errorCode !== 0) {
                reject(new Error(`Command failed, \`${cmd} ${args.join(' ')}\`, ${stack.join(', ')}`));
            }
            resolve(
                response
                    .replace(/[\r?\n|[\] ']/g, '') // Remove all chars, new lines and empty space
                    .trim()
                    .split(',')
            );
        });
    });
}
```

Reply:
Ill update so it only runs the specified command. I believe we don't use the http call for a reason internally.